### PR TITLE
`Learning analytics`: Fix missing name of attachment lecture units on student dashboard

### PR DESCRIPTION
**Pull request by FelixTJDietrich**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).

### Motivation and Context

The GAD course uses a lot of attachment units and the titles are missing:
Screenshot 2024-06-11 at 00 01 30

### Description

Use title of the attachment and include it in the DTO for display.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with student analytics course dashboard enabled
- 1 Attachment lecture unit in the course
- 1 Competency in the course

1. Log in to Artemis
2. Navigate to course with dashboard
3. Expand the competency of the attachment unit
4. Verify that the title is no longer missing

#### Performance Review
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

no change

### Screenshots

Fixed:
Screenshot 2024-06-11 at 00 00 03

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved data retrieval in lecture unit information by handling potential null values and including attachments for more comprehensive results.

### CODE REVIEW

**Review by bensofficial — APPROVED** [review]
Code change looks good to me. Also tested on TS3 and the names are all visible now. Thanks for the fix 🙌.

**Review by rstief — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]

File: src/main/java/de/tum/in/www1/artemis/repository/metrics/LectureUnitMetricsRepository.java [hunk 1]
```diff
@@ -27,8 +27,9 @@ public interface LectureUnitMetricsRepository extends JpaRepository<LectureUnit,
      * @return the lecture unit information for all lecture units in the course
      */
     @Query("""
-            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.LectureUnitInformationDTO(lu.id, lu.lecture.id, lu.name, lu.releaseDate, TYPE(lu))
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.LectureUnitInformationDTO(lu.id, lu.lecture.id, COALESCE(lu.name, a.name), lu.releaseDate, TYPE(lu))
             FROM LectureUnit lu
+            LEFT JOIN Attachment a ON a.attachmentUnit.id = lu.id
```
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/in/www1/artemis/repository/metrics/LectureUnitMetricsRepository.java:32]:
```suggestion
                LEFT JOIN Attachment a ON a.attachmentUnit.id = lu.id
```

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]

**Review by bensofficial — APPROVED** [review]
Reapprove

**Review by MaximilianAnzinger — APPROVED** [review]

**Review by pzdr7 — APPROVED** [review]
Tested on TS3. Created a lecture unit with type attachment and linked it to a competency. Did not encounter any missing names.

**Review by bassner — APPROVED** [review]

**Pull request merged**